File: dotnet/platform/workspace/typescript/libs/protocol/system/src/pull/pullrequest.ts
import { Pull } from '@allors/data/system';

export class PullRequest {
  public pulls?: Pull[];

  constructor(fields?: Partial<PullRequest>) {
    Object.assign(this, fields);
  }

  public toJSON() {
    return {
      p: this.pulls,
    };
  }
}

File: dotnet/system/protocol/allors.protocol.json/api/pull/pullrequest.cs
// <copyright file="PullRequest.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Protocol.Json.Api.Pull
{
    using Data;

    public class PullRequest
    {
        /// <summary>
        /// List of Pulls
        /// </summary>
        public Pull[] l { get; set; }

        /// <summary>
        /// Procedure
        /// </summary>
        public Procedure p { get; set; }
    }
}
